rlp/rlpgen/types.go

```diff
@@ -113,6 +113,16 @@ func isByte(typ types.Type) bool {
 	return ok && basic.Kind() == types.Uint8
 }
 
+// isNamedWithBasicUnderlying checks whether 'typ' is a named type with an underlying basic type.
+func isNamedWithBasicUnderlying(typ types.Type) bool {
+	named, ok := typ.(*types.Named)
+	if !ok {
+		return false
+	}
```

Comment:
```suggestion
func hasBasicUnderlying(named *types.Named) bool {
```

The switch statement from which this is called has already established that it's a `*types.Named`.